src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -307,6 +307,10 @@ def validate_authentication_type(namespace, formula=None):
                 "incorrect usage for authentication-type 'password': "
                 "[--admin-username USERNAME] --admin-password PASSWORD")
 
+        # Respect user's provided saved secret name for password authentication
+        if namespace.saved_secret:
+            namespace.admin_password = "[[{}]]".format(namespace.saved_secret)
```

Comment:
Yup `[[VALUE]]` is the construct expected from the REST endpoint so that it understands that the value needs to be searched in lab's secrets store.